python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/resources.py

```diff
@@ -374,14 +375,29 @@ def _replicate(
 
             end_time = time.time()
 
+            has_asset_def: bool = bool(context and context.has_assets_def)
```

Comment:
based on the type annotation of the argument, context will never be `None`, right? in which case I don't think you need the `context and`?